module/shop/shop_voucher.py

```diff
@@ -0,0 +1,230 @@
+        shop_grid = ButtonGrid(
+            origin=(463, origin_y), delta=(156, delta_y), button_shape=(99, 99), grid_shape=(5, row), name='SHOP_GRID')
+        return shop_grid
+
+    @cached_property
+    def shop_voucher_items(self):
+        """
+        Returns:
+            ShopItemGrid:
+        """
+        shop_grid = self.shop_grid
+        shop_voucher_items = ShopItemGrid(
+            shop_grid,
+            templates={}, amount_area=(60, 74, 96, 95),
+            price_area=(52, 132, 132, 162))
+        shop_voucher_items.load_template_folder('./assets/shop/voucher')
+        shop_voucher_items.load_cost_template_folder('./assets/shop/cost')
+        shop_voucher_items.similarity = 0.85
+        shop_voucher_items.cost_similarity = 0.5
+        shop_voucher_items.price_ocr = PRICE_OCR
+        return shop_voucher_items
+
+    def shop_items(self):
+        """
+        Shared alias name for all shops,
+        so to use  @Config must define
+        a unique alias as cover
+
+        Returns:
+            ShopItemGrid:
+        """
+        return self.shop_voucher_items
+
+    def shop_currency(self):
+        """
+        Ocr shop voucher currency
+        Then return voucher count
+
+        Returns:
+            int: voucher amount
+        """
+        self._currency = OCR_SHOP_VOUCHER.ocr(self.device.image)
+        logger.info(f'Voucher: {self._currency}')
+        return self._currency
+
+    def shop_interval_clear(self):
+        """
+        Clear interval on select assets for
+        shop_buy_handle
+        """
+        super().shop_interval_clear()
+        self.interval_clear(SHOP_BUY_CONFIRM_SELECT)
+        self.interval_clear(SHOP_BUY_CONFIRM_AMOUNT)
+        self.interval_clear(SHOP_BUY_CONFIRM_VOUCHER)
+
+    def shop_buy_handle(self, item):
+        """
+        Handle shop_voucher buy interface if detected
+
+        Args:
+            item: Item to handle
+
+        Returns:
+            bool: whether interface was detected and handled
+        """
+        if self.appear(SHOP_BUY_CONFIRM_SELECT, offset=(20, 20), interval=3):
+            self.shop_buy_select_execute(item)
+            self.interval_reset(SHOP_BUY_CONFIRM_SELECT)
+            return True
+        if self.appear(SHOP_BUY_CONFIRM_AMOUNT, offset=(20, 20), interval=3):
+            self.shop_buy_amount_execute(item)
+            self.interval_reset(SHOP_BUY_CONFIRM_AMOUNT)
+            return True
+        if self.appear_then_click(SHOP_BUY_CONFIRM_VOUCHER, offset=(20, 20), interval=3):
+            return True
+
+        return False
+
+    def run(self):
+        """
+        Run Voucher Shop
+        """
+        # Base case; exit run if filter empty
+        if not self.shop_filter:
+            return
+
+        # When called, expected to be in
+        # correct Voucher Shop interface
+        logger.hr('Voucher Shop', level=1)
+        self.wait_until_voucher_appear()
+
+        # Execute buy operations
+        while 1:
```

Comment:
Add before loop:
```python
MATERIAL_SCROLL.set_top(main=self)
```

Reply:
Can do, guess I can update `MedalShop` with the same since I took it from there. So if it didn't have it here, then likely not there either

Reply:
Done, added to both